### PR TITLE
Extend settings_runtime_commit to support all commits

### PR DESCRIPTION
**Pull request by hno-wires**
This replaces settings_runtime_commit() with a wrapper for settings_commit_subtree(), extending the functionality to support global commit and nested subtree commits.

Fixes #51703

### CODE REVIEW

**Comment by carlescufi** [comment]
@hno-addiva thanks!
See https://docs.zephyrproject.org/latest/contribute/guidelines.html#commit-guidelines

**Comment by de-nordic** [comment]
Please also add commit with release notes entry mentioning the Zephyr issue being fixed.

**Comment by de-nordic** [comment]
@hno-addiva ping?

**Comment by rakons** [comment]
It seems that some settings related tests are failing.

**Review by rakons — CHANGES_REQUESTED** [review]
NAK as the tests are failing. It seems that settings_runtime_commit functionality was different than settings_commit_subtree.

**Comment by teburd** [comment]
@mniestroj please take a look

File: subsys/settings/src/settings_runtime.c [hunk 1]
```diff
@@ -58,17 +58,5 @@ int settings_runtime_get(const char *name, void *data, size_t len)
...
-	if (!ch) {
-		return -EINVAL;
-	}
-
-	if (ch->h_commit) {
-		return ch->h_commit();
-	} else {
-		return 0;
-	}
+	return settings_commit_subtree(name);
```
[the remark this answers is not in the document]
Comment (mniestroj) [reply subsys/settings/src/settings_runtime.c:61]: > I also thought of proposing it to be dropped as useless.

I was not using `settings_runtime` recently and actually this was my first view into `settings_runtime_commit()` implementation. But my first impression was that it should be dropped.

This is also the reason why I referenced 94451b22fcc6a068de2a2982b202772c6bbefae8, which provides no explanation why it was introduced in the first place. And from that commit we can see that the only user of this API were some test cases.

Just sharing my concerns here, I don't have strong opinion on that. I think also that we lack some sort of guidelines for when to use settings vs settings_runtime and what should be the use cases. At least that is not clear to me, even after using both, I am still not certain that I am doing it in a valid way.

**Comment by MaureenHelm** [comment]
@de-nordic ping

**Comment by de-nordic** [comment]
> @de-nordic ping

There isn't that much interest in the change, and it seems that issue has not been raised by other developers so far.

By the comment https://github.com/zephyrproject-rtos/zephyr/pull/51705#discussion_r1116020316
> But not going to work more on this at the moment. The project where we was using Zephyr is unfortunately on hold (but not due to Zephyr fault). We will surely come back to Zephyr in future.
It seems that currently the fix will not be pursued by the author either. Test have also not been altered to work with the enhancement provided here.

If this is still considered need enhancement issue and anyone finds this work to be important improvement, please add Enhance request to Zephyr issues, describing what is currently done wrong and how proposed solution fixes that, and follow it with implementation (if any).

@MaureenHelm  Closing due to lack of interest at the moment.

**Pull request closed without merging**